modules/atomic-widgets/base/atomic-widget-base.php

```diff
@@ -24,20 +24,23 @@ public function __construct( $data = [], $args = null ) {
 
 	public function get_atomic_controls() {
 		$controls = $this->define_atomic_controls();
+		$schema = static::get_props_schema();
+
+		// Validate the schema only in the Editor.
+		static::validate_schema( $schema );
 
-		return $this->get_valid_controls( $controls );
+		return $this->get_valid_controls( $schema, $controls );
 	}
 
-	private function get_valid_controls( array $controls ): array {
+	private function get_valid_controls( array $schema, array $controls ): array {
```

Comment:
Don't you think better approach will be to call the `get_props_schema` method?

Reply:
It's in order to avoid calculating the schema many times per widget